### DISCUSSION TITLE
Java custom metrics without editing code

### DISCUSSION THREAD

**Discussion by Syberia5910** [open-telemetry/opentelemetry-java-instrumentation#8543] [Q&A]
Hi,

I want to add custom metrics with the java agents without editing code (for the moment we don't want impacting projets). For example, we have a generic class that call external service in a particular protocol and we wan't to count number of call and response time. 

I see that I can create an extensions but I don't find some examples for creating some buckets for response time. 

Is the extension is the good way for creating custom mesurment or there is other thing ?

I want to mesure datasource's response time too and redis's calls. I see java agent has redis dans dbcp2 module but I think those module is just for trace and not for mesurement. Is it wright ? 
 
Thanks

**Comment by laurit** [comment]
Bytecode instrumentation with extension is intended to be used when you can't modify the source code. There are some samples with metrics in https://opentelemetry.io/docs/instrumentation/java/manual/#metrics Currently we don't a have redis instrumentation that would gather metrics.
Although your eventual goal might be to write an extension and not modify your application I'd suggest you'd start from trying out the metics apis inside your application. Doing the modifications in the applications will hopefully give you an understanding of what you need to instrument inside your application.
